lib/redis.js

```diff
@@ -106,16 +104,43 @@ module.exports = CoreObject.extend({
       })
       .then(function() {
         return client.set(redisKey, value);
-      })
+      });
   },
 
   _updateRecentUploadsList: function(keyPrefix, revisionKey) {
     var client = this._client;
-    return client.lpush(keyPrefix, revisionKey);
+    var score = new Date().getTime();
+    return client.zadd(keyPrefix, score, revisionKey);
+  },
+
+  _activeRevision: function(keyPrefix) {
+    var currentKey = keyPrefix + ':current';
+    return this._client.get(currentKey);
   },
 
   _trimRecentUploadsList: function(keyPrefix, maxEntries) {
+    /**
+     * look at all the keys being removed
```

Comment:
agreed, plan is to remove these before merge
but since we had a few conversations about the algorithm to use I thought I'd left them in during the review